docs/csharp/tutorials/default-interface-members-versions.md

```diff
@@ -0,0 +1,80 @@
+
+Notice the highlighted portion of the test. That cast from `SampleCustomer` to `ICustomer` is necessary. The `SampleCustomer` class doesn't need to provide an implementation for `ComputeLoyaltyDiscount`; that's provided by the `ICustomer` interface. However, the `SampleCustomer` class doesn't inherit members from its interfaces. That rule hasn't changed. In order to call any method declared and implemented in the interface, the variable must be the type of the interface, `ICustomer` in this example.
+
+## Provide parameterization
+
+That's a good start. But, the default implementation is too restrictive. Many consumers of this system may choose different thresholds for number of purchases, a different length of membership, or a different percentage discount. You can provide a better upgrade experience for more customers by providing a way to set those parameters. Let's add a static method that sets those three parameters controlling the default implementation:
+
+[!code-csharp[VersionTwoImplementation](~/samples/csharp/tutorials/default-interface-members-versions/finished/customer-relationship/ICustomer.cs?name=SnippetLoyaltyDiscountVersionTwo)]
+
+There's many new capabilities shown in that small code fragment. Interfaces can now include static members, including fields and methods. Different access modifiers are also enabled. In the previous example, the additional fields are private, and the new method is public. Any of the modifiers are allowed on interface members.
+
+Applications that use the general formula for computing the loyalty discount, but different parameters don't need to provide a custom implementation; they can set the arguments through a static method. For example, the following code sets a "customer appreciation" that rewards any customer with more than one month's membership:
+
+[!code-csharp[SetLoyaltyThresholds](~/samples/csharp/tutorials/default-interface-members-versions/finished/customer-relationship/Program.cs?name=SnippetSetLoyaltyThresholds)]
+
+## Extend the default implementation
+
+The code you've added so far has provided a convenient implementation for those scenarios where users want something like the default implementation, or to provide an unrelated set of rules. For a final feature, let's refactor the code a bit to enable scenarios where users may want to build on the default implementation. 
+
+Consider a startup that wants to attract new customers. They offer a 50% discount off a new customer's first order. Otherwise, existing customers get the standard discount. A bit of refactoring in the `ICustomer` interface makes this scenerio easy:
+
+[!code-csharp[VersionTwoImplementation](~/samples/csharp/tutorials/default-interface-members-versions/finished/customer-relationship/ICustomer.cs?name=SnippetFinalVersion)]
```

Comment:
I think this adds complexity without any benefit. I expected to see the `ComputeLoyaltyDiscount` method fully implemented here but it instead calls out to the shared static member. Why is that? It wasn't implied in this scenario that the user is creating multiple ICustomer implementations and would want to share functionality between them all (`DefaultLoyaltyDiscount`).

After reading this again and the next paragraph, I think that this code is being written by the library author? However, the introduction of this area talks about things from the users perspective, so I expected to see the user implement some overriding logic here.

Reply:
I reworked the section to explain the scenario better.  I do want to show a utility method added to the interface.